### DISCUSSION TITLE
NixOS: No extensions being shown as installed

### DISCUSSION THREAD

**Discussion by aftix** [FreshRSS/FreshRSS#6578] [Q&A] [answered]
### Describe the bug

I have freshrss installed in /var/lib/freshrss . I have copied several extensions directories into /var/lib/freshrss/extensions . The web UI is showing no extensions as installed.

### To Reproduce

1. Copy extension directories from https://github.com/FreshRSS/Extensions to ./extensions in the freshrss install location
2. Go to the `<freshrss base url>/i/?c=extension` page
3. No extensions are installed
![image](https://github.com/FreshRSS/FreshRSS/assets/4008299/aae7bf10-d759-4272-9279-6925b9f68a04)

### Expected behavior

I expect extensions to show up as installed on this page as the installation instructions were followed.

### FreshRSS version

1.24.1

### Environment information

- NixOS unstable, commit version e97b3e4186bcadf0ef1b6be22b8558eab1cdeb5d
- SQLite 3.45.3 
- php 8.3.8
- Installed into /var/lib/freshrss, run using systemd services for the updater/phpfpm/etc on the host
 (https://github.com/NixOS/nixpkgs/blob/57d6973abba7ea108bac64ae7629e7431e0199b6/nixos/modules/services/web-apps/freshrss.nix shows the service configurations)
- Run behind a reverse proxy of nginx 1.26.1
- Web UI viewed on linux with firefox 127.0

The configuration (including copying the extensions) can be found in https://github.com/aftix/cfg/blob/main/host/opt/www/rss.nix

**Comment by math-GH** [comment]
Could you please show us the content of the `/var/lib/freshrss/extensions` directory?

**Comment by aftix** [comment]
![image](https://github.com/FreshRSS/FreshRSS/assets/4008299/ffea3929-63ca-45f3-82c2-005a6152b786)
These are directories copied from various extension sources, including FreshRss/Extensions

**Comment by math-GH** [comment]
and what is the content of `xExtension-CustomCSS` directory?

**Comment by aftix** [comment]
the directory contents are:
```
LICENSE  README.md  configure.phtml  desktop_resolution.png  extension.php  i18n  metadata.json  mobile_resolution.png  static
```

It is copied directly from https://github.com/FreshRSS/Extensions/tree/master/xExtension-CustomCSS

**Comment by math-GH** [comment]
ok, the directory looks okay.

**Comment by Alkarex** [comment]
> the directory contents are

Please show the details of that folder with `ls -alh`. In particular, are those files or symlinks?

**Comment by aftix** [comment]
![image](https://github.com/FreshRSS/FreshRSS/assets/4008299/3863306b-b929-4446-8858-14c3dd16014c)
This is the contents of xExtension-CustomCSS,
![image](https://github.com/FreshRSS/FreshRSS/assets/4008299/04af921b-3780-43b9-a962-f89a7a05ea29)
and these are the contents of the extensions directory.

The directories are being copied using `cp -LR` (aka `cp -R --dereference`), so there should be no symlinks anywhere in the extensions directory

**Comment by Alkarex** [comment]
Are you sure this `www` user is the correct one?
Please show the `ls -alh` of your `freshrss/data/` folder

**Comment by aftix** [comment, marked as answer]
I have found the issue - on NixOS freshrss is installed in a `/nix/store/....` path, and the `/var/lib/freshrss` path just contains some data, moving the extensions to the nix path's extensions directory works as expected
